src/inverted_index.c

```diff
@@ -226,32 +226,35 @@ ENCODER(encodeRawDocIdsOnly) {
 
 #define NUM_TINYENC_MASK 0x07  // This flag is set if the number is 'tiny'
 
+#define NUM_ENCODING_COMMON_TYPE_TINY           0
+#define NUM_ENCODING_COMMON_TYPE_FLOAT          1
+#define NUM_ENCODING_COMMON_TYPE_POSITIVE_INT   2
+#define NUM_ENCODING_COMMON_TYPE_NEG_INT        3
+
 typedef struct {
-  uint8_t deltaEncoding : 2;
-  uint8_t zero : 2;
-  uint8_t valueByteCount : 3;
-  uint8_t sign : 1;
+  uint8_t deltaEncoding : 3;
```

Comment:
So the delta now is 3 bytes? Wouldn't this gives us better flexibility when choosing to start a  new block on the inverted index?

Reply:
Also please notice that changing the encoding will break loading rdb from before 2.0, but I think its OK to say that its not longer supported, we do need to specify it in the release notes though and  remove this entire code that handles loading old rdb's (on another PR).
@gkorland @K-Jo FYI

Reply:
We used to have up to 4 bytes (2^2 excluding zero) to store delta, now it will be 7 (2^3 including zero)
So practically we can remove this block which added a new block when delta exceeds 4 bytes
Since now the delta limit would be 2^56-1
```
if (delta > UINT32_MAX) {
    blk = InvertedIndex_AddBlock(idx, docId);
    delta = 0;
  }
```

Reply:
Actually, support for zero delta is only added for numeric encoder/decoder, and it doesn't affect others,
so checking not to exceed the delta limit is still needed for other encoders,
and for numeric encoder the limit is currently practically unreachable.